### DISCUSSION TITLE
WinUI 3 desktop support?

### DISCUSSION THREAD

**Discussion by watsonwelch** [dotnet/System.Graphics#42] [Q&A]
Hello! This is a really interesting project. What are your thoughts on WinUI 3 desktop (Win32) support? I know many of your existing abstractions (Win2D, SkiaSharp, etc.) have various challenges associated with WinUI 3. Thanks for your consideration!

**Comment by charlesroddie** [comment]
What are the challenges @watsonwelch ? How could WinUI possibly not support SharpDX, Win2D, and SkiaSharp?

**Reply by watsonwelch** [reply]
Thanks for the reply! The challenges as I understand them are:

- SharpDX stop being maintained in 2019 so it's unclear if it's still safe to be used as a dependency (.NET 5 isn't listed as an officially supported platform, for example).
- When I wrote the question it was unclear if Win2D was going to be officially supported on WinUI 3 for desktop (Win32) since it's historically been UWP-only — however it looks like [Win2D for WinUI 3 desktop support was just released a couple days ago](https://www.nuget.org/packages/Microsoft.Graphics.Win2D), so this would presumably be the most obvious path.
- SkiaSharp support for WinUI 3 I believe is in development but not yet released, and the maintainer mentions on [this thread](https://github.com/mono/SkiaSharp/issues/1298) that "GPU support is not working though because ANGLE does not support WinUI yet"

**Reply by charlesroddie** [reply]
Great thanks for the additional info. I think Win2D will be the natural choice.
